### PR TITLE
{AKS} Extend `--data-collection-settings` with new field "streams"

### PR DESCRIPTION
**Pull request by ganga1980**
**Related command**

**Description**
Extend DataCollectionSettings  JSON object with streams field and update API version for the DCR & DCR-A to version 2022-06-01

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — APPROVED** [review]
The change looks good to me. Please also sync the changes to aks-preview.

Is there any plan to gather these monitoring addon-related operations into an SDK? I'm concerned that manual changes will inevitably miss something.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -7366,10 +7366,15 @@ def create_new_cluster_with_monitoring_aad_auth(self, resource_group, resource_g
              self.cmd(get_cmd, checks=[
                 self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.namespaces[0]', f'kube-system')
             ])
-
+             self.cmd(get_cmd, checks=[
+                self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.streams[0]', f'Microsoft-ContainerLogV2')
+            ])
+             self.cmd(get_cmd, checks=[
+                self.check('properties.dataFlows[0].streams[0]', f'Microsoft-ContainerLogV2')
+            ])
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7374]: It is suggested to combine these checks in one command implementation, e.g., 

```python
self.cmd(get_cmd, checks=[
                self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.namespaces[0]', f'kube-system'),
                self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.streams[0]', f'Microsoft-ContainerLogV2'),
                self.check('properties.dataFlows[0].streams[0]', f'Microsoft-ContainerLogV2'),
            ])
```

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -7365,6 +7365,9 @@ def create_new_cluster_with_monitoring_aad_auth(self, resource_group, resource_g
             ])
              self.cmd(get_cmd, checks=[
                 self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.namespaces[0]', f'kube-system')
+            ])
+             self.cmd(get_cmd, checks=[
+                self.check('properties.dataSources.extensions[0].extensionSettings.dataCollectionSettings.enableContainerLogV2', f'true')
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7370]: Queued [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=75185750&view=results) (case `test_aks_create_with_monitoring_aad_auth_msi_with_datacollectionsettings`) to validate the change.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (ganga1980) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py]: fixed.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**